### PR TITLE
python3Packages.sphinxcontrib-wavedrom: init at 3.0.4

### PR DESCRIPTION
**Pull request by fsagbuya**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/sphinxcontrib-wavedrom/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    wavedrom
+    xcffib
+    cairosvg
+  ];
+
+  # No tests included
+  doCheck = false;
+
+  pythonImportsCheck = [ "sphinxcontrib.wavedrom" ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/sphinxcontrib-wavedrom/default.nix:35]:
```suggestion

  pythonNamespaces = [ "sphinxcontrib" ];

```
Could you please add `pythonNamespaces`?
Without it, the sphinx version upgrade may break this package.
see more details, https://github.com/NixOS/nixpkgs/pull/259350

File: pkgs/development/python-modules/sphinxcontrib-wavedrom/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+, sphinx
+, wavedrom
+, xcffib
+, cairosvg
+}:
+
+buildPythonPackage rec {
+  pname = "sphinxcontrib-wavedrom";
+  version = "3.0.4";
+  format = "pyproject";
```
Comment (natsukium) [inline pkgs/development/python-modules/sphinxcontrib-wavedrom/default.nix:15]:
```suggestion
  pyproject = true;
```
We have a new parameter for pyproject.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 264321` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.sphinxcontrib-wavedrom</li>
    <li>python310Packages.sphinxcontrib-wavedrom.dist</li>
    <li>python311Packages.sphinxcontrib-wavedrom</li>
    <li>python311Packages.sphinxcontrib-wavedrom.dist</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM, I'll merge it after CI passes.

**Review by Janik-Haag — APPROVED** [review]

**Comment by natsukium** [comment]
@ofborg build python311Packages.sphinxcontrib-wavedrom

**Comment by Janik-Haag** [comment]
you don't have to do a extra ofborg build if the darwin tests seem stuck, it's just we don't have enough darwin build infra

**Comment by natsukium** [comment]
If the package name in the commit is an alias, ofborg will not start the build. So unfortunately I have to call it manually.
python3Packages is a commonly used alias, but I recommend the current python311Packages.

**Comment by Janik-Haag** [comment]
hm interesting,  the linux stuff build before you pinged ofborg :thinking:

**Pull request merged**